client/tests/simulator_tests.rs

```diff
@@ -83,6 +85,66 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random_max_size() {
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: 4096,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[..resp.random_bytes.size as usize];
+
+    // We should have received less than what the client implementation supports.
+    assert!(random_slice.len() <= TpmtHa::UNION_SIZE);
+
+    // We should have received minimum size of SHA256 (minimum mandatory digest size).
+    assert!(random_slice.len() >= TPM2_SHA256_DIGEST_SIZE as usize);
+
+    // Print for test inspection.
+    trace!("Got random sequence: {:?}", random_slice);
```

Comment:
Maybe just me, but I don't think I'd take a crate dependency for debug prints in tests. You only see the prints when the test fails anyways, and I think that's a desirable trait, myself. Why are we adding this `trace` stuff?

Reply:
Original idea was to add tracing so it can be disabled/enabled at will - but adding messages to the assert messages is fine here. Fixed 👍 🙂